dbms/src/Common/MPMCQueue.h

```diff
@@ -353,6 +375,8 @@ class MPMCQueue
             notifyNext(reader_head);
             return Result::OK;
         }
+        if (is_timeout)
+            return Result::TIMEOUT;
```

Comment:
why move to here?

Reply:
Consider a case that `wait` is timeout, and at the meantime, there is a notifiy on the condition variable, in the original implementation, the notification is actually lost, so we double check the condition even if `wait` return false.